polysemy-plugin/test/InsertSpec.hs

```diff
@@ -0,0 +1,31 @@
+module InsertSpec where
+
+import Polysemy.Reader (Reader)
+import Polysemy.State (State)
+import Polysemy.Internal (Sem)
+import Polysemy.Internal (insertAt)
+import Polysemy
+import Test.Hspec
+
+insertSome :: ∀ e1 e2 i r a . Sem (e1 ': e2 ': r) a -> Sem (e1 ': e2 ': Reader i ': r) a
+insertSome = insertAt @2 @'[Reader i]
+
+-- insertSome2 :: forall e1 e2 e4 i r a. Sem (State Int ': r) a -> Sem (State Int ': Reader i ': r) a
+-- insertSome2 = insertAfter @(State Int) @'[Reader i]
+
+-- insertAfterTest2 :: forall i r a. Sem (Reader i ': Reader Int ': r) a -> Sem (Reader i ': State Int ': Reader Int ':  r) a
+-- insertAfterTest2 = insertAfter @(Reader i) @'[State Int]
+
+-- insertAtTest2 ::
+--   forall e1 e2 e4 i r a.
+--   Member e2 (e1 : e2 : State Int : e4 : Reader i : r) =>
+--   Sem (e1 ': e2 ': r) a ->
+--   Sem (e1 ': e2 ': State Int ': e4 ': Reader i ': r) a
+-- insertAtTest2 = insertAt @2 @'[State Int, e4, Reader i]
+
+
+spec :: Spec
+spec = parallel $ do
+  describe "insert" $ do
+    it "insert" $ do
+      1 `shouldBe` 1
```

Comment:
If we can bear the _headache_, we should have tests for compile-time error messages like the `polysemy-plugin` tests do.